From 0297d8078b7c26b030fd8a723a220febbec1cc7c Mon Sep 17 00:00:00 2001
From: Arthur Zamarin <arthurzam@gentoo.org>
Date: Sun, 23 Oct 2022 21:18:27 +0300
Subject: [PATCH] fix include memory with USE_VULKAN=OFF

Those header files are using std::shared_ptr from memory header even
when USE_VULKAN is OFF, so we must include them unconditionally.

Signed-off-by: Arthur Zamarin <arthurzam@gentoo.org>
---
 src/qmplay2/LibASS.hpp     | 2 +-
 src/qmplay2/QMPlay2OSD.hpp | 3 +--
 2 files changed, 2 insertions(+), 3 deletions(-)

diff --git a/src/qmplay2/LibASS.hpp b/src/qmplay2/LibASS.hpp
index 70d29193..0d4eba7c 100644
--- a/src/qmplay2/LibASS.hpp
+++ b/src/qmplay2/LibASS.hpp
@@ -23,6 +23,7 @@
 #include <QByteArray>
 #include <QList>
 
+#include <memory>
 #include <set>
 
 class Settings;
@@ -35,7 +36,6 @@ struct ass_renderer;
 struct ass_image;
 
 #ifdef USE_VULKAN
-#include <memory>
 namespace QmVk {
 class BufferPool;
 }
diff --git a/src/qmplay2/QMPlay2OSD.hpp b/src/qmplay2/QMPlay2OSD.hpp
index cf2291c3..38e08fb1 100644
--- a/src/qmplay2/QMPlay2OSD.hpp
+++ b/src/qmplay2/QMPlay2OSD.hpp
@@ -25,14 +25,13 @@
 #include <QRect>
 
 #include <functional>
+#include <memory>
 #include <vector>
 #include <mutex>
 
 #ifdef USE_VULKAN
 #   include <QVector4D>
 
-#   include <memory>
-
 namespace QmVk {
 
 class Buffer;
